appformer-kogito-api/pom.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2020 Red Hat, Inc. and/or its affiliates.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>uberfire-parent</artifactId>
+    <groupId>org.uberfire</groupId>
+    <version>7.34.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>appformer-kogito-api</artifactId>
```

Comment:
reading the code, assuimng your explanation, this module should be named appformer-state-control-api?

Reply:
I'd prefer not having a dedicated module for state control since we are not sure how will be growing.

Reply:
ok, understand, then still think we need to explain properly purpose of this module somewhere

Reply:
@pefernan I like all of your refactoring; thank-you..... other than the name of this new module.

It has nothing to do with _kogito_ itself and is indeed, just a module with zero/minimal dependencies on other things to prevent the kogito specific implementation module dragging in unnecessary libraries. Can you come up with a better name that does not mention kogito!?!? (`appformer-client-api` is super-nice BUT obviously cannot be used as it has a ton of other dependencies you don't want in kogito). `appformer-collaboration-api` is reasonable - but limits its future use. `appformer/uberfire-services/uberfire-services-client` is available!